### ISSUE TITLE
UnnecessaryParentheses should be ignored for bit operations

### ISSUE THREAD

**Issue by Mygod** [arturbosch/detekt#1929]
## Expected Behavior of the rule

Example:

```
        val mask = 256 - (1 shl (i * 8 + 8 - prefixSize))
        return (a[i].toInt() and mask) == (b[i].toInt() and mask)
```

## Context

Bitwise operations in C-like languages have confusing priorities so people usually add parentheses for everything.
This sample code also does not trigger warnings in latest Android Studio (3.6 alpha).

detekt 1.0.1

**Comment by 3flex** [comment]
1.1.0 has a change in behaviour for this rule - it should only flag issues that are also flagged by Kotlin plugin for IntelliJ (see #1890).

When 1.1.0 is released please see if the new behaviour helps.

If not it's still unlikely that we'll make changes and instead suggest suppressing the rule in these cases, since there were a few issues with this rule in the past and too many edge cases, so we'll now just use same logic as Kotlin itself does.

**Comment by schalkms** [comment]
This has been fixed by 61f1734.
We won’t invest more time in this rule, since detekt complies with IntelliJ’s rule.

**Comment by Mygod** [comment]
In 1.1.0 this issue is still there. Also IntelliJ/Android Studio does not report this.

**Comment by 3flex** [comment]
Sorry I wasn't clearer earlier.

IntelliJ has intentions and inspections. Inspections show warnings or errors, intentions are used to change code but don't show a warning or error. Your example code is picked up by the intention detection in IntelliJ as expected, and detekt does the same:

![image](https://user-images.githubusercontent.com/170028/67002788-5ba47080-f128-11e9-966b-02e2a5b649f0.png)

**Comment by Mygod** [comment]
Oh I see. Thank you!
